pkg/queue/stats_reporter.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type Measurement int
+
+// TODO(@mrmcmuffinz): Need to move this to a more appropriate place.
+// TODO(@mrmcmuffinz): Need to get more appropriate names.
+const (
+	// ReportingPeriod interval of time for reporting.
+	ReportingPeriod = 10 * time.Second
+	// RequestCountM number of requests received since last Stat (approximately QPS).
+	RequestCountM Measurement = iota
+	// AverageConcurrentRequestsM average number of requests currently being handled by this pod.
+	AverageConcurrentRequestsM
+	// LameDuckM indicates this Pod has received a shutdown signal.
+	LameDuckM
+)
+
+// TODO(@mrmcmuffinz): Need to move this to a more appropriate place.
+var (
+	measurements = []*stats.Float64Measure{
+		RequestCountM: stats.Float64(
+			"request_count_total",
```

Comment:
This is actually OPS (operations per second) because the stat reporting period is 1 second.  There is another kind of request count which accumulates over the lifetime of the process.  `request_count_total` sounds like that, but it's not.

I suggest renaming this as `operations_per_second` and opening a TODO (linked to an issue) to process the stats to make sure this remains accurate with reporting periods other than 1 second.